loki_role/tasks/loki.yaml

```diff
@@ -0,0 +1,81 @@
+---
+- name: updating cache of ubuntu
+  ansible.builtin.apt:
+    update_cache: yes
+
+- name: Install unzip, bzip2, xz-utils, and zstd
+  ansible.builtin.apt:
+    name:
+      - unzip
+      - bzip2
+      - xz-utils
+      - zstd
+    state: present
+
+- name: Download Binaries for loki
+  ansible.builtin.get_url:
+    url: "https://github.com/grafana/loki/releases/download/v{{ loki_version }}/loki-linux-amd64.zip"
+    dest: /home/ubuntu/
+    mode: '755'
+
+- name: Unarchive a file 
+  ansible.builtin.unarchive:
+    src: "/home/ubuntu/loki-linux-amd64.zip"
+    dest: /home/ubuntu/
+    remote_src: yes
+
+
+- name: Add the user 'loki'
+  ansible.builtin.user:
+    name: loki
+    state: present
+    system: true
+
+- name: Copy file with owner and permissions
+  ansible.builtin.copy:
+    src: /home/ubuntu/loki-linux-amd64 
+    dest: /usr/local/bin/loki
+    owner: loki
+    group: loki
+    mode: '755'
+    remote_src: true
+
+
+- name: Creating directory for config file
+  ansible.builtin.file:
+    path: "{{ item }}" 
+    state: directory
+    owner: loki
+    group: loki
+    mode: '755'
+    recurse: true
+  loop: 
+    - /etc/loki/logs
+    - /data/loki
+
+- name: Creating Config file
+  ansible.builtin.file:
+    path: /etc/loki/loki-local-config.yaml 
+    state: touch
+
+
+- name: Copying the config file content
+  ansible.builtin.template:
+    src: templates/loki-config.j2
+    dest: /etc/loki/loki-local-config.yaml
```

Comment:
Have we tested this if it is working fine or not. I think you haven't add values to jinja template till now directly copy pasting the jinja template.